src/llvm-lower-handlers.cpp

```diff
@@ -150,7 +150,16 @@ static bool lowerExcHandlers(Function &F) {
                 EnterDepth[CI] = Depth++;
             else if (Callee == leave_func || Callee == leave_noexcept_func) {
                 LeaveDepth[CI] = Depth;
-                Depth -= cast<ConstantInt>(CI->getArgOperand(1))->getLimitedValue();
+                if (auto change = dyn_cast<ConstantInt>(CI->getArgOperand(1)))
+                    Depth -= change->getLimitedValue();
+                else if (auto Phi = dyn_cast<PHINode>(CI->getArgOperand(1))) {
+                    //This should really do a dataflow analysis but assuming worst case means that we will always have enough space
```

Comment:
```suggestion
                    // XXX: This should really do a dataflow analysis but assuming worst case means that we will always have enough space
```

Reply:
Coould we have an IR test for this?

Reply:
There is one :). I was also talking with @topolarity and @vtjnash that we should just do what that pass does at codegen time and remove it.